storage.py

```diff
@@ -98,6 +98,31 @@ def zipup(self, in_name, out_name):
         # SessionException will be raised from run() in icommands.py
         self.session.run("ibun", None, '-cDzip', '-f', out_name, in_name)
 
+    def unzip(self, zip_file_path):
+        """
+        run iRODS ibun command to unzip files
+        :param zip_file_path: path of the zipped file to be unzipped
```

Comment:
yeah, the problem we're running into is when a zipped file is unzipped next to an aggregation, the metadata files for the aggregation get added to django.  Unzipping files into a single folder and only adding the folder contents to django is the solution that arose.

Reply:
@sblack-usu I have approved this PR, but then another thought come into my mind. I am thinking whether it is worthwhile to make this unzip method a more generic method by providing a target_unzip_path as an optional input argument with a default value set as you did for ```unzipped_folder``` currently. I am also thinking ```_get_nonexistant_path()```` may not be needed but instead just override the target path if it already exists. Again I have not got a chance to see how this is being used, so I may misunderstand the intention, but just some thoughts for your consideration.

Reply:
The thought occurred to me as well to provide a way to override.  However, I decided against implementing it at this time because we are currently not using it.  It can be easily implemented when there is a use case.

As for the get_non_existent_path() method, I expect it to be used very little because there most likely won't be a conflict.  In the case of a conflict I thought it best to not overwrite files without the user's consent.  Considering it's both not likely to happen often and it's much more work to implement a system notifying and handling feedback from the user, I believe this is the best solution with the time constraints we have.

Reply:
OK. You can merge this when you are ready then.

Reply:
You have me second guessing myself now.  Some of the tests in the hydroshare project failed with my changes.  Turns out there are some tests out there expecting an exception to be thrown if overwriting files.  (http://ci.hydroshare.org:8080/job/hydroshare-pull-requests/1741/testReport/hs_core.tests/TestResourceFileFolderOprsAPI/test_resource_file_folder_oprs/)

I'm going to ask the development channel to see if there is good reason for this, or if I should update the tests to look for the new outcome.